test/cocotb/osdtestlib/soc_interface.py

```diff
@@ -0,0 +1,36 @@
+import cocotb
+from cocotb.triggers import Timer
+from cocotb.clock import Clock
+from cocotb.result import TestFailure
+from cocotb.triggers import RisingEdge
+
+from osdtestlib.debug_interconnect import RegAccess
+
+
+class TraceGenerator:
+    """
+    Mimics CPU signals to generate stimuli for connected debug modules
+    """
+
+    @cocotb.coroutine
+    def trigger_event(self, dut, trace_id, trace_value):
+        """Sets the trace* signals of a debug module so that a debug event will
+           be generated
+        Args:
+            dut:              device under test.
+            trace_id:         ID of the trace event.
+            trace_value:      value of the trace event.
+        Returns:
+            -
+        """
+
+        # Trigger the generation of a new debug event packet
+        dut.fifo_valid       <= 1
```

Comment:
Where is that fifo_valid signal coming from? It's not part of the module port list (https://github.com/opensocdebug/hardware/blob/osd-next/modules/stm/common/osd_stm.sv#L19)?